ambry-api/src/main/java/com/github/ambry/config/RouterConfig.java

```diff
@@ -578,6 +580,20 @@ public class RouterConfig {
   @Default("15*1000")
   public final long routerNotFoundCacheTtlInMs;
 
+  /**
+   * Toggles metadata cache improvement in NonBlockingRouter.
+   * If true, metadata chunk of a composite blob is cached.
+   * If false, there is no caching of metadata chunks.
+   */
+  @Config(ROUTER_METADATA_CACHE_ENABLED)
+  public final boolean routerMetadataCacheEnabled;
```

Comment:
Looks like setting the cache entry size to 0 also disables the cache by evicting every entry. But, may be it is better to config to be more explicit and avoid touching the cache.